### PR TITLE
PSR-1163 Hide incomplete members from lists and disable payments links for 1 incomplete member

### PR DESCRIPTION
**Pull request by B-Savage**

### CODE REVIEW

File: app/controllers/nonsipp/employercontributions/EmployerContributionsMemberListController.scala [hunk 1]
```diff
@@ -62,26 +62,8 @@ class EmployerContributionsMemberListController @Inject()(
 
   def onPageLoad(srn: Srn, page: Int, mode: Mode): Action[AnyContent] = identifyAndRequireData(srn) {
     implicit request =>
-      val memberMap = request.userAnswers.map(MembersDetailsPages(srn))
```
Comment (kberec) [inline app/controllers/nonsipp/employercontributions/EmployerContributionsMemberListController.scala:65]: Tagging @ccatosdevelopment for awareness, please check if this is still ok with your changes going on in parallel with this change

File: app/pages/nonsipp/memberdetails/MemberDetailsPage.scala [hunk 2]
```diff
@@ -48,5 +50,27 @@ object MembersDetailsPage {
 
   implicit class MembersDetailsOps(ua: UserAnswers) {
     def membersDetails(srn: Srn): Map[String, NameDOB] = ua.map(MembersDetailsPages(srn))
+
+    def membersOptionList(srn: Srn): List[Option[NameDOB]] = {
```
Comment (kberec) [inline app/pages/nonsipp/memberdetails/MemberDetailsPage.scala:54]: 👍

File: app/controllers/nonsipp/memberdetails/SchemeMembersListController.scala [hunk 3]
```diff
@@ -17,7 +17,7 @@
 package controllers.nonsipp.memberdetails
 
 import services.PsrSubmissionService
-import pages.nonsipp.memberdetails.SchemeMembersListPage
```
Comment (kberec) [inline app/controllers/nonsipp/memberdetails/SchemeMembersListController.scala:20]: Are any changes needed in SchemeMembersListControllerSpec to check the completed pages filtering?

File: app/controllers/nonsipp/employercontributions/EmployerContributionsMemberListController.scala (hunk 1 above)
Comment (ccatosdevelopment) [reply app/controllers/nonsipp/employercontributions/EmployerContributionsMemberListController.scala:65]: Thanks @kberec, this doesn't overlap with any of the changes I've made so far, but I'll keep it in mind if I make any further changes to this file.

File: app/controllers/nonsipp/memberdetails/SchemeMembersListController.scala (hunk 3 above)
Comment (B-Savage) [reply app/controllers/nonsipp/memberdetails/SchemeMembersListController.scala:20]: Added a test for this

**Review by kberec — DISMISSED** [review]

**Review by kberec — APPROVED** [review]

**Pull request merged**